addons/account_peppol/models/res_partner.py

```diff
@@ -0,0 +1,485 @@
+    'LI': {'9936': 'vat'},
+    'LT': {'9937': 'vat'},
+    'LU': {'9938': 'vat'},
+    'LV': {'0218': 'company_registry', '9939': 'vat'},
+    'MC': {'9940': 'vat'},
+    'ME': {'9941': 'vat'},
+    'MK': {'9942': 'vat'},
+    'MT': {'9943': 'vat'},
+    # Do not add the vat for NL, since: "[NL-R-003] For suppliers in the Netherlands, the legal entity identifier
+    # MUST be either a KVK or OIN number (schemeID 0106 or 0190)" in the Bis 3 rules (in PartyLegalEntity/CompanyID).
+    'NL': {'0106': None, '0190': None},
+    'NO': {'0192': 'l10n_no_bronnoysund_number'},
+    'NZ': {'0088': 'company_registry'},
+    'PL': {'9945': 'vat'},
+    'PT': {'9946': 'vat'},
+    'RO': {'9947': 'vat'},
+    'RS': {'9948': 'vat'},
+    'SE': {'0007': 'company_registry'},
+    'SI': {'9949': 'vat'},
+    'SK': {'9950': 'vat'},
+    'SM': {'9951': 'vat'},
+    'TR': {'9952': 'vat'},
+    'VA': {'9953': 'vat'},
+}
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    account_peppol_is_endpoint_valid = fields.Boolean(
+        string="PEPPOL endpoint validity",
+        help="The partner's EAS code and PEPPOL endpoint are valid",
+        compute="_compute_account_peppol_is_endpoint_valid", store=True,
+        copy=False,
+    )
+    account_peppol_validity_last_check = fields.Date(
+        string="Checked on",
+        help="Last Peppol endpoint verification",
+        compute="_compute_account_peppol_is_endpoint_valid", store=True,
+        copy=False,
+    )
+    account_peppol_verification_label = fields.Selection(
+        selection=[
+            ('not_verified', 'Not verified yet'),
+            ('not_valid', 'Not valid'),  # does not exist on Peppol at all
+            ('not_valid_format', 'Cannot receive this format'),  # registered on Peppol but cannot receive the selected document type
+            ('valid', 'Valid'),
+        ],
+        string='Peppol endpoint validity',
+        compute='_compute_account_peppol_verification_label',
+        copy=False,
+    )  # field to compute the label to show for partner endpoint
+
+    def _auto_init(self):
+        """Create columns `account_peppol_is_endpoint_valid` and `account_peppol_validity_last_check`
+        to avoid having them computed by the ORM on installation.
+        """
+        if not column_exists(self.env.cr, 'res_partner', 'account_peppol_is_endpoint_valid'):
+            create_column(self.env.cr, 'res_partner', 'account_peppol_is_endpoint_valid', 'boolean')
+        if not column_exists(self.env.cr, 'res_partner', 'account_peppol_validity_last_check'):
+            create_column(self.env.cr, 'res_partner', 'account_peppol_validity_last_check', 'timestamp')
+        return super()._auto_init()
+
+    # -------------------------------------------------------------------------
+    # HELPER METHODS
+    # -------------------------------------------------------------------------
+
+    @api.model
+    def _get_participant_info(self, edi_identification):
```

Comment:
As discussed, this needs to be adapted for NAPTR migration that is ongoing.
See https://github.com/odoo/odoo/pull/232483 .